src/platform/plugins/shared/discover/public/components/discover_grid/discover_grid.tsx

```diff
@@ -48,6 +48,7 @@ export const DiscoverGrid: React.FC<DiscoverGridProps> = ({
       query,
       updateESQLQuery: onUpdateESQLQuery,
       setExpandedDoc,
```

Comment:
```suggestion
      setExpandedDoc: renderDocumentView ? setExpandedDoc : undefined,
```
This was the area I meant in my original comment.